tools/grid/src/GridController.ts

```diff
@@ -119,28 +131,43 @@ export class GridController<T extends HTMLElement>
         } else if (typeof gap === 'function') {
             this._gap = gap;
         }
+        /* c8 ignore next 2 */
+        if (typeof padding === 'string') {
+            this._padding = () => padding;
+        } else if (typeof padding === 'function') {
+            this._padding = padding;
+        }
     }
 
     public update({
         elements,
         itemSize,
         gap,
+        padding,
     }: {
         elements: () => T[];
         itemSize: ItemSize | (() => ItemSize);
         gap?: string | (() => string);
+        padding?: string | (() => string);
     }): void {
         this.rovingTabindexController.update({ elements });
-        this.applyLayout(itemSize, gap);
+        this.applyLayout(itemSize, gap, padding);
         const contentRect = this.host.getBoundingClientRect();
         this.measureDirectionLength(contentRect);
     }
 
     protected measureDirectionLength(contentRect: DOMRect): void {
         const gap = this.gap ? parseFloat(this.gap) : 0;
-        this.rovingTabindexController.directionLength = Math.floor(
-            (contentRect.width - gap) / (this.itemSize.width + gap)
-        );
+        const padding = this.padding ? parseFloat(this.padding) : 0;
+        let contentBoxWidth =
+            contentRect.width - padding * 2 - this.itemSize.width;
+        let directionLength = 0;
+        while (contentBoxWidth > 0) {
+            directionLength += 1;
+            contentBoxWidth = contentBoxWidth - (gap + this.itemSize.width);
+        }
```

Comment:
Would this be equivalent?

```suggestion
        const directionLength = Math.floor(contentBoxWidth / (gap + this.itemSize.width));
```

Reply:
I'm sure there's a smarter math here. This comes from the idea that if there at `n` items in a row, then there are `n - 1` gaps, so you can't just say `gap + itemWidth`.  So maybe it's clearer with:
```js
        // account for there being one less gap than items in a row
        contentBoxWidth = contentBoxWidth - this.itemSize.width;
        const directionLength = Math.floor(contentBoxWidth / (gap + this.itemSize.width)) + 1;
```